modules/reactor/src/meshgenerators/HexagonMeshTrimmer.C

```diff
@@ -0,0 +1,574 @@
+                              p0(1) - p1(1),
+                              p1(0) - p0(0),
+                              -(p0(0) * (p0(1) - p1(1)) + (p1(0) - p0(0)) * p0(1)));
+    }
+    // Node 1 is on the removal side while node 0 is on the retaining side
+    else if (side_id_1_in && (side_node_0_remove != side_node_1_remove))
+    {
+      node_list_flag[std::distance(
+          node_list.begin(), std::find(node_list.begin(), node_list.end(), side_id_1))] = true;
+      const Point p0 = *mesh.node_ptr(side_id_0);
+      const Point p1 = *mesh.node_ptr(side_id_1);
+
+      node_list_point[std::distance(node_list.begin(),
+                                    std::find(node_list.begin(), node_list.end(), side_id_1))] =
+          twoLineIntersection(bdry_pars[0],
+                              bdry_pars[1],
+                              bdry_pars[2],
+                              p0(1) - p1(1),
+                              p1(0) - p0(0),
+                              -(p0(0) * (p0(1) - p1(1)) + (p1(0) - p0(0)) * p0(1)));
+    }
+  }
+
+  // move nodes
+  for (unsigned int i = 0; i < node_list.size(); i++)
+  {
+    // Only one node in trimmed region
+    // This means the node is on both the trimming boundary and the original external boundary.
+    // In order to keep the shape of the original external boundary, the node is moved along the
+    // original external boundary.
+    if (node_list_flag[i])
+      *(mesh.node_ptr(node_list[i])) = node_list_point[i];
+    // Two nodes in trimmed region, only one is moved
+    // This means the node is in the middle of the trimming boundary.
+    // Just move it along the normal direction of the trimming line.
+    else
+    {
+      const Real x0 = (*(mesh.node_ptr(node_list[i])))(0);
+      const Real y0 = (*(mesh.node_ptr(node_list[i])))(1);
+      (*(mesh.node_ptr(node_list[i])))(0) =
+          (bdry_pars[1] * (bdry_pars[1] * x0 - bdry_pars[0] * y0) - bdry_pars[0] * bdry_pars[2]) /
+          (bdry_pars[0] * bdry_pars[0] + bdry_pars[1] * bdry_pars[1]);
+      (*(mesh.node_ptr(node_list[i])))(1) =
+          (bdry_pars[0] * (-bdry_pars[1] * x0 + bdry_pars[0] * y0) - bdry_pars[1] * bdry_pars[2]) /
+          (bdry_pars[0] * bdry_pars[0] + bdry_pars[1] * bdry_pars[1]);
+    }
+  }
+
+  // Delete the block
+  for (auto elem_it = mesh.active_subdomain_elements_begin(block_id_to_remove);
+       elem_it != mesh.active_subdomain_elements_end(block_id_to_remove);
+       elem_it++)
+    mesh.delete_elem(*elem_it);
+  mesh.contract();
+  mesh.find_neighbors();
+  // Delete zero volume elements
+  for (auto elem_it = mesh.elements_begin(); elem_it != mesh.elements_end(); elem_it++)
+  {
+    if (MooseUtils::absoluteFuzzyEqual((*elem_it)->volume(), 0.0))
+    {
+      for (unsigned int i = 0; i < (*elem_it)->n_sides(); i++)
+      {
+        if ((*elem_it)->neighbor_ptr(i) != nullptr)
+        {
+          boundary_info.add_side((*elem_it)->neighbor_ptr(i),
```

Comment:
this isnt covered. Any meshes in the reactor module which would trigger that?

Reply:
Typical reactor module meshes might not trigger this. 
lineRemover() has been designed for more general trimming purposes, which would lead to zero volume elements. I would like to keep these lines for future expansion.